src/execution/operator/aggregate/physical_window.cpp

```diff
@@ -459,34 +472,85 @@ static void SortCollectionForPartition(PhysicalWindowOperatorState &state, Bound
 	local_sort_state.Initialize(global_sort_state, *state.buffer_manager);
 
 	// sink collection chunks into row format
-	const idx_t chunk_count = sort_collection.ChunkCount();
+	const idx_t chunk_count = over.ChunkCount();
 	for (idx_t i = 0; i < chunk_count; i++) {
-		auto &sort_chunk = *sort_collection.Chunks()[i];
-		auto &payload_chunk = *input.Chunks()[i];
-		local_sort_state.SinkChunk(sort_chunk, payload_chunk);
+		auto &over_chunk = *over.Chunks()[i];
+		auto &payload_chunk = *payload.Chunks()[i];
+
+		// Extract the hash partition, if any
+		if (hashes) {
+			auto &hash_chunk = *hashes->Chunks()[i];
+			auto hash_size = hash_chunk.size();
+			auto hash_data = FlatVector::GetData<hash_t>(hash_chunk.data[0]);
+			idx_t bin_size = 0;
+			for (idx_t i = 0; i < hash_size; ++i) {
+				if ((hash_data[i] & hash_mask) == hash_bin) {
+					sel.set_index(bin_size++, i);
+				}
+			}
+
+			// Flush the partition chunks if we would overflow
+			if (over_partition.size() + bin_size > STANDARD_VECTOR_SIZE) {
+				local_sort_state.SinkChunk(over_partition, payload_partition);
```

Comment:
This appears to be uncovered, could you try to add a test case that triggers this?

Reply:
Amusingly this is the piece of code I was profiling for 4 days!  But yes, it takes a test case with multiple partitions that are over 2kb. I'll throw something together.